src/library/dlgtrackinfo.cpp

```diff
@@ -613,9 +613,20 @@ void DlgTrackInfo::saveTrack() {
     slotSpinBpmValueChanged(spinBpm->value());
     static_cast<void>(updateKeyText()); // discard result
 
+    // Don't update if nothing changed.
+    if (m_trackRecord == m_pLoadedTrack->getRecord() &&
+            m_pBeatsClone == m_pLoadedTrack->getBeats()) {
```

Comment:
In case metadata or beats have been changed elsewhere we'd already have updated here as well, so this scheck is safe IMO (i.e. does **not** undo those changes by re-applying this dialog's (unchanged) record and beats).

Admittedly, since the update is triggered by signals there is a slight chance users revert their own manual changes, but I'd consider that a "user error" because working on metadata in multiple places at once (at inhuman high rate) may inevitably create incosistencies.

Any thoughts?